src/utils/locale.ts

```diff
@@ -61,57 +66,41 @@ function createLocaleNumberFormatter(locale: string): Intl.NumberFormat {
 
 export function delocalizeNumberString(numberString: string, locale: string): string {
   return sanitizeExponentialNumberString(numberString, (nonExpoNumString: string): string => {
-    if (nonExpoNumString) {
-      const groupSeparator = getGroupSeparator(locale);
-      const decimalSeparator = getDecimalSeparator(locale);
-      const minusSign = getMinusSign(locale);
-
-      const splitNumberString = nonExpoNumString.split("");
-      const decimalIndex = splitNumberString.lastIndexOf(decimalSeparator);
-
-      const delocalizedNumberString = splitNumberString
-        .map((value, index) => {
-          if (value === groupSeparator || (value === decimalSeparator && index !== decimalIndex)) {
-            return "";
-          }
-          return value;
-        })
-        .reduce((string, part) => string + part)
-        .replace(decimalSeparator, ".")
-        .replace(minusSign, "-")
-        .replace(/[^0-9\-\.]/g, ""); // remove everything except numbers, "-", and "."
+    const delocalizedNumberString = nonExpoNumString
+      .replace(getMinusSign(locale), "-")
+      .replace(getGroupSeparator(locale), "")
+      .replace(getDecimalSeparator(locale), ".")
+      .replace(allDecimalsExceptLast, "")
+      .replace(everythingExceptNumbersDecimalsAndMinusSigns, "");
 
-      return isNaN(Number(delocalizedNumberString)) ? nonExpoNumString : delocalizedNumberString;
-    }
-    return nonExpoNumString;
+    return isValidNumber(delocalizedNumberString) ? delocalizedNumberString : nonExpoNumString;
   });
 }
 
 export function getGroupSeparator(locale: string): string {
   const formatter = createLocaleNumberFormatter(locale);
-  const parts = formatter.formatToParts(1234567.8);
+  const parts = formatter.formatToParts(1234567);
```

Comment:
Is having the `.8` necessary for this calculation?

Reply:
No, which is why I removed it. That function only needs to find the group separator part. Now there will be less parts to iterate through in that function, FWIW.

Reply:
Oh woops, my bad, I was reading that diff in the opposite way.  I guess that means I'm going color blind?!?! lol